xapian-bindings/java/java.i

```diff
@@ -189,6 +189,64 @@ namespace Xapian {
     bool hasNext() const { return Xapian::iterator_valid(*self); }
 }
 
+%typemap(javainterfaces) Xapian::MSetIterator "Iterable<MSetIterator.MatchDescriptor>";
+%typemap(javaimports) Xapian::MSetIterator "import java.util.Iterator;"
+%typemap(javacode) MSetIterator %{
+  public Iterator<MatchDescriptor> iterator() {
+      Iterator<MatchDescriptor> iterator = new Iterator <MatchDescriptor>() {
+          @Override
+          public boolean hasNext() {
+              return MSetIterator.this.hasNext();
+          }
+          @Override
+          public MatchDescriptor next() {
+              MSetIterator thisIterator = MSetIterator.this;
+              long rank = thisIterator.getRank();
+              long docId = thisIterator.getDocId();
+              double weight = thisIterator.getWeight();
+              int percent = thisIterator.getPercent();
+              Document doc = thisIterator.getDocument();
+              MatchDescriptor descriptor = new MatchDescriptor(docId, rank, percent, weight, doc);
+	      thisIterator.next();
+              return descriptor;            
+          }
+       };
+  
+      return iterator;
+  }
+
+  public class MatchDescriptor {
+      private long docId,rank;
+      private int percent;
+      private double weight;
+      private Document document;
+
+      public MatchDescriptor(long docId, long rank, int percent, double weight, Document document) {
+          this.docId = docId;
+          this.rank = rank;
+          this.percent = percent;
+          this.weight = weight;
+          this.document = document;
+      }
+
+      public long getDocId() { return docId; }
+      public void setDocId(long docId) { this.docId = docId; }
+
+      public long getRank() { return rank; }
+      public void setRank(long rank) { this.rank = rank; }
+
+      public int getPercent() { return percent; }
+      public void setPercent(int percent) { this.percent = percent; }
+
+      public double getWeight() { return weight; }
+      public void setWeight(double weight) { this.weight = weight; }
+
+      public Document getDocument() { return document; }
+      public void setDocument(Document document) { this.document = document; } 
+  }
```

Comment:
I would suggest omitting the setters here.  They aren't needed for the expected use of the iterator, and if we don't have setters there's more scope for changing the implementation - for example, `MatchDescriptor` could hold an `MSetIterator` and just call that to lazily fetch the things which are actually asked for.